server/src/middleware/verify-token.middleware.ts

```diff
@@ -0,0 +1,47 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { errorHandler } from '../utils';
+import { HttpMethods, Separators, StatusCode, messages } from '../constants';
+import { JWT_SECRET } from '../public.env';
+import { UserEntity } from '../resources/user/user.interfaces';
+
+const authorizationCallBack = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): Promise<Response | undefined | void> => {
+  if (req.method === HttpMethods.OPTIONS) {
+    return next();
+  }
+
+  const auth = req.headers.authorization as string;
+
+  if (!auth) {
+    return res.status(StatusCode.UNAUTHORIZED).json({ message: messages.notAuthorized });
+  }
+
+  // authorization = `Bearer ${token}`
+  const [tokenType, token] = auth.split(Separators.SPACE);
+
+  if (tokenType !== 'Bearer') {
+    return res.status(StatusCode.FORBIDDEN).json({ message: messages.invalidToken });
+  }
+
+  const user = jwt.verify(token, JWT_SECRET) as UserEntity;
```

Comment:
This line causes app to crash in case when I specify invalid JWT token. To test, you may try this one:
```
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzZXNzaW9uSWQiOiIyODk5ZmMxMC0zODlhLTExZTktOTFkNi0wZDZhMTM3MDVhZmQiLCJ1c2VySWQiOiJkZTQ5MDU3MC0zODk5LTExZTktYWNiOC0yMWRmOGI0NjBhODIiLCJwZXJtaXNzaW9
ucyI6W10sImlhdCI6MTU1MTA1NzAzNn0.3kiTyTyxchl9gPj1Kl4P1aOYITKQ_CEepwCsnSjB338
```

Reply:
The same error appears if I specify invalid JWT token

Reply:
Done.